### PR TITLE
Platform-3221: set https protocol based on Fastly headers

### PR DESCRIPTION
**Pull request by themech**

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 4 info

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebRequest.php#](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#): File "WebRequest.php" has 1455 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WebRequest.php#L38](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L38): Class "WebRequest" has 55 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WebRequest.php#L54](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L54): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [WebRequest.php#L197](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L197): Remove this unused "$stdPort" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WebRequest.php#L337](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L337): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WebRequest.php#L1078](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L1078): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WebRequest.php#L1078](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L1078): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [WebRequest.php#L1285](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L1285): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WebRequest.php#L1365](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L1365): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR][MINOR] [WebRequest.php#L197](https://github.com/Wikia/app/blob/7ec1e1435c3d8d81eb4e4ef2a4c04a9a1402a452/includes/WebRequest.php#L197): Immediately return this expression instead of assigning it to the temporary variable "$proto". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: includes/WebRequest.php [hunk 1]
```diff
@@ -171,7 +171,9 @@ public static function detectServer() {
 			}
 			$host = $parts[0];
 			if ( $parts[1] === false ) {
-				if ( isset( $_SERVER['SERVER_PORT'] ) ) {
+				if ( !empty( $_SERVER['HTTP_FASTLY_SSL'] ) ) {
+					$port = 443;
+				} elseif ( isset( $_SERVER['SERVER_PORT'] ) ) {
 					$port = $_SERVER['SERVER_PORT'];
 				} // else leave it as $stdPort
```
Comment (WikiaTech) [inline includes/WebRequest.php:178]: INFO Move this trailing comment on the previous empty line. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS139)

[project-id]: wikia:app

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 9 major
* ![MINOR][MINOR] 4 minor
* ![INFO][INFO] 4 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WebRequest.php#](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#): File "WebRequest.php" has 1457 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WebRequest.php#L38](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L38): Class "WebRequest" has 55 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WebRequest.php#L54](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L54): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [WebRequest.php#L199](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L199): Remove this unused "$stdPort" local variable. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
1. ![MAJOR][MAJOR] [WebRequest.php#L339](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L339): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WebRequest.php#L1080](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L1080): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WebRequest.php#L1080](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L1080): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [WebRequest.php#L1287](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L1287): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MAJOR][MAJOR] [WebRequest.php#L1367](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L1367): Add curly braces around the nested statement(s). [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS121)
1. ![MINOR][MINOR] [WebRequest.php#L199](https://github.com/Wikia/app/blob/07d5b66f7fa2cf6d8afba14bcaa67ef9f78691f0/includes/WebRequest.php#L199): Immediately return this expression instead of assigning it to the temporary variable "$proto". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1488)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 32 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L42](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L42): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): The Cyclomatic Complexity of this function "execute" is 83 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): This function "execute" has 533 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L465](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L465): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L486](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L486): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L649](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L649): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L672](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L672): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. ![MAJOR][MAJOR] [WebRequest.php#](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/includes/WebRequest.php#): File "WebRequest.php" has 1457 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WebRequest.php#L38](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/includes/WebRequest.php#L38): Class "WebRequest" has 55 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WebRequest.php#L54](https://github.com/Wikia/app/blob/52d6d6e449fdd8e71c6be51c899b7aab81d34058/includes/WebRequest.php#L54): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by Grunny — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 32 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L42](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L42): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): The Cyclomatic Complexity of this function "execute" is 83 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): This function "execute" has 533 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L465](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L465): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L486](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L486): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L649](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L649): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L672](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L672): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. ![MAJOR][MAJOR] [WebRequest.php#](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/includes/WebRequest.php#): File "WebRequest.php" has 1457 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WebRequest.php#L38](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/includes/WebRequest.php#L38): Class "WebRequest" has 55 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WebRequest.php#L54](https://github.com/Wikia/app/blob/5b8c810078e7b4cb83b54a82498f80b71e8e745d/includes/WebRequest.php#L54): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 32 issues
* ![MAJOR][MAJOR] 16 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 10 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L42](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L42): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): The Cyclomatic Complexity of this function "execute" is 83 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L226](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L226): This function "execute" has 533 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS138)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L465](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L465): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L486](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L486): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L649](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L649): Refactor this code to not nest more than 4 "if", "for", "while", "switch" and "try" statements. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS134)
1. ![MAJOR][MAJOR] [WikiFactoryLoader.php#L672](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/extensions/wikia/WikiFactory/Loader/WikiFactoryLoader.php#L672): Add a "case default" clause to this "switch" statement. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS131)
1. ![MAJOR][MAJOR] [WebRequest.php#](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/includes/WebRequest.php#): File "WebRequest.php" has 1457 lines, which is greater than 1000 authorized. Split it into smaller files. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS104)
1. ![MAJOR][MAJOR] [WebRequest.php#L38](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/includes/WebRequest.php#L38): Class "WebRequest" has 55 methods, which is greater than 20 authorized. Split it into smaller classes. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1448)
1. ![MAJOR][MAJOR] [WebRequest.php#L54](https://github.com/Wikia/app/blob/8519de63e5ce3ae16acc579d659fff9a5d0dcf19/includes/WebRequest.php#L54): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)

[project-id]: wikia:app
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**